cli/command/secret/create.go

```diff
@@ -46,28 +50,26 @@ func runSecretCreate(dockerCli command.Cli, options createOptions) error {
 	client := dockerCli.Client()
 	ctx := context.Background()
 
-	var in io.Reader = dockerCli.In()
-	if options.file != "-" {
-		file, err := system.OpenSequential(options.file)
-		if err != nil {
-			return err
-		}
-		in = file
-		defer file.Close()
+	if options.driver != "" && options.file != "" {
+		return errors.Errorf("When using secret driver secret data must be empty")
```

Comment:
Could this be handled by the external driver? IIUC, with this change the secret's value is taken from the external store, so

```bash
$ docker secret create -d external mysecret
```

Doesn't really "create" a secret, but "associates" the external secret with a secret in Docker.

Do we need to keep the option open to have external secret stores create new secrets from the Docker CLI? i.e.

```bash
$ echo "new secret" | docker secret create -d external mysecret -
```

If so, this validation should be done by the driver (just like currently checking for _empty_ values is done in the backend)